### PR TITLE
PLAT-10480 : fix spring starter's extended healthCheck

### PR DESCRIPTION
**Pull request by symphony-soufiane**
### Ticket
[PLAT-10480](https://perzoinc.atlassian.net/browse/PLAT-10480)

### Description
This is PR is related to [PLAT-10480](https://perzoinc.atlassian.net/browse/PLAT-10480).
The extended health check was not working in Spring starter. The issue comes from the api clients configuration that injects a pod api client instead of agent's one to SystemApi that provides healthCheck. 
The bdk core is not impacted as its configuration is already correct.

### Dependencies
List the other pull requests that should be merged before/along this one.

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [x] Filled properly the description and dependencies, if any
- [ ] Unit tests updated or added
- [ ] Javadoc added or updated
- [ ] Updated the documentation in [docs folder](../docs)

### CODE REVIEW

**Comment by symphony-soufiane** [comment]
No unit tests added as the change only covered configuration.

**Comment by symphony-thibault** [comment]
> No unit tests added as the change only covered configuration.

Not even possible to test in https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/blob/master/symphony-bdk-spring/symphony-bdk-core-spring-boot-starter/src/test/java/com/symphony/bdk/spring/SymphonyBdkAutoConfigurationTest.java?

**Comment by symphony-youri** [comment]
One idea could be to check that the apiclient is the service is the same as the agent apiclient (in the test Thibault mentioned where we can access injected beans)

**Review by symphony-youri — APPROVED** [review]

File: symphony-bdk-spring/symphony-bdk-core-spring-boot-starter/src/test/java/com/symphony/bdk/spring/SymphonyBdkAutoConfigurationTest.java [hunk 1]
```diff
@@ -225,4 +226,17 @@ void shouldNotInitializeBotSessionWhenOboOnly() {
...
+  @Test
+  void systemApiClientShouldTargetAgentBasePath() {
+    final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+        .withUserConfiguration(SymphonyBdkMockedConfiguration.class)
+        .withConfiguration(AutoConfigurations.of(SymphonyBdkAutoConfiguration.class));
+
+    contextRunner.run(context -> {
+      String actualBasePath = context.getBeansOfType(SystemApi.class).get("systemApi").getApiClient().getBasePath();
+      assertThat(actualBasePath).isEqualTo("/agent");
+    });
```
Comment (symphony-thibault) [inline symphony-bdk-spring/symphony-bdk-core-spring-boot-starter/src/test/java/com/symphony/bdk/spring/SymphonyBdkAutoConfigurationTest.java:239]: 👍

**Pull request merged**